scylla-cql/src/frame/value.rs

```diff
@@ -51,6 +51,150 @@ pub enum MaybeUnset<V> {
     Set(V),
 }
 
+/// Represents timeuuid (uuid V1) value
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
```

Comment:
I'm wondering if it would be beneficial to users to derive `Hash` to, so this type is easier to use in hash maps. wdyt?

Reply:
It's good that you mentioned that.

Answering your question - we shouldn't derive `Hash` as our `PartialOrd`/`Ord` custom logic masks off 7th bit msbyte. Scylla/Cassandra both behave in the same way.

The same applies to deriving `Eq`/`PartialEq`. We should implement them in a way so the following property holds: `x.cmp(&y) == Ordering::Equal <=> x == y`.

Here is an exemplary test case showing that, as of now, this property doesn't hold:
```rust
#[test]
fn timeuuid_hash() {
    let x = CqlTimeuuid::from_str("00000000-0000-1000-8080-808080808080").unwrap();
    let y = CqlTimeuuid::from_str("00000000-0000-2000-8080-808080808080").unwrap();

    let cmp_res = x.cmp(&y);
    assert_eq!(Ordering::Equal, cmp_res);

    assert_eq!(x, y); // this panics

    let compute_hash = |x: &CqlTimeuuid| {
        let mut hasher = DefaultHasher::new();
        x.hash(&mut hasher);
        hasher.finish()
    };
    assert_eq!(compute_hash(&x), compute_hash(&y)); // this panics
}
```

I'll implement the `Hash` trait as well.